src/goto-instrument/code_contracts.cpp

```diff
@@ -129,11 +130,25 @@ void code_contractst::check_apply_invariants(
       t->location_number > loop_end->location_number)
       loop_end = t;
 
-  // see whether we have an invariant
+  // see whether we have an invariant and a variant
   exprt invariant = static_cast<const exprt &>(
     loop_end->get_condition().find(ID_C_spec_loop_invariant));
+  exprt variant = static_cast<const exprt &>(
+    loop_end->get_condition().find(ID_C_spec_decreases));
   if(invariant.is_nil())
-    return;
+  {
+    if(variant.is_nil())
+      return;
+    else
+    {
+      invariant = true_exprt();
+      log.warning()
+        << "The loop at " << loop_head->source_location.as_string()
+        << " does not have a loop invariant, but has a loop variant. "
```

Comment:
Please can you be consistent with naming?  Personally I don't like "loop variant" but I care most about it being consistent.  SO if you are going to call it a loop variant in the help text and comments then maybe it should be `__CPROVER_loop_variant`.

Reply:
Thanks for your suggestion. As of now, the concrete syntax for loop variants is `__CPROVER_decreases(...)`. However, "decreases" is not a noun (or at least, it would be awkward to use it as a noun). This is why I currently use the phrase "loop variants" in the code/comments to refer to the loop variants/ranking functions. I will discuss this with Saswat.

Reply:
@martin-cs I misunderstood your comment - I thought you were suggesting changing all function names and comments in the code base so that we never mention "loop variants", but now I believe you are just referring to the warning issued to a user. Yes, I will change the warning message to something like "The loop at ... does not have a loop invariant, but has a decreases-clause." In the code base, however, I will keep using the phrase "loop variants" in comments and function names.

Reply:
So, I am very bad at using consistent terminology and it causes confusion.  My comments, here and earlier, are intended to try to help you avoid a similar fate.

I would /suggest/ that you use one terminology for *everything*; the annotation, the user error reporting, in the comments, in the code, everywhere.  Either call them ranking functions, or loop variants, or ... whatever name you want.  Personally I feel that loop variant can cause confusion with loop invariants but that is just my view.  I think it is far more important to be consistent than it is to pick one particular name.

Reply:
OK. Thanks for the clarification! I will update my comments and function names in the code.

Reply:
I've just updated all function names, comments, and warning messages in the code such that the phrase "DECRAESES clause" is used in place of "loop variants."